dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -103,6 +118,111 @@
         return iterate(query);
     }
 
+    enum OP {equals,not_equals,like,not_like,contains,doesnt_contain,exists,doesnt_exist,matches,doesnt_match;}
+    
+    @Override
+    public Iterator<Item> findByMetadataQuery(Context context, List<String> query_field, List<String> query_op, List<String> query_val, List<String> collection_ids, String regexClause) throws SQLException {
+    	Criteria criteria = createCriteria(context, Item.class, "item");
+    	
+		List<UUID> uuids = new ArrayList<UUID>();
+		for(String s: collection_ids) {
+			try {
+				uuids.add(UUID.fromString(s));
+			} catch (IllegalArgumentException e) {
+				log.warn("Invalid collection UUID: " + s);
+			}
+		}
+    	if (!uuids.isEmpty()){
+			DetachedCriteria dcollCriteria = DetachedCriteria.forClass(Collection.class, "coll");
+        	dcollCriteria.setProjection(Projections.property("coll.id"));
+        	dcollCriteria.add(Restrictions.eqProperty("coll.id", "item.owningCollection"));
+			dcollCriteria.add(Restrictions.in("coll.id", uuids));
+			criteria.add(Subqueries.exists(dcollCriteria));
+    	}
+    	
+        int index = Math.min(query_field.size(), Math.min(query_op.size(), query_val.size()));
+        StringBuilder sb = new StringBuilder();
+
+        for(int i=0; i<index; i++) {
+        	OP op = OP.valueOf(query_op.get(i));
+        	if (op == null) {
+        		log.warn("Skipping Invalid Operator: " + query_op.get(i));
+        		continue;
+        	}
+        	
+        	if (op == OP.matches || op == OP.doesnt_match) {
+        		if (regexClause.isEmpty()) {
+            		log.warn("Skipping Unsupported Regex Operator: " + query_op.get(i));
+            		continue;
+        		}
+        	}
+        	
+        	
+        	String schema = "";
+        	String element = "";
+        	String qualifier = "";
+        	boolean allFields = query_field.get(i).equals("*");
+        	String[] parts = query_field.get(i).split("\\.");
+        	if (parts.length>0) {
+        		schema = parts[0];
+        	}
+        	if (parts.length>1) {
+        		element = parts[1];
+        	}
+        	if (parts.length>2) {
+        		qualifier = parts[2];
+        	}
+        	
+        	DetachedCriteria subcriteria = DetachedCriteria.forClass(MetadataValue.class,"mv");
+        	subcriteria.add(Property.forName("mv.dSpaceObject").eqProperty("item.id"));
+        	subcriteria.setProjection(Projections.property("mv.dSpaceObject"));
+        	
+        	if (!allFields) {
+        		DetachedCriteria  fieldCriteria = DetachedCriteria.forClass(MetadataField.class,"mfr");
+        		fieldCriteria.add(Property.forName("mv.metadataField").eqProperty("mfr.id"));
+        		fieldCriteria.setProjection(Projections.property("mfr.id"));
```

Comment:
Instead of splitting up the schema metatafield string & using it in a query, perhaps lookup the metadatafield itself. The MetadataField & MetadataSchema rows are cached across sessions so a query on MetadataField doesn't even hit the database but rather the cache.
It also results in a lot less code.
Make sure to perform the lookup from the service side.